### PR TITLE
Bug 1693308 Use highest seen versions in AMO stats

### PR DESCRIPTION
**Pull request by jklukas**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1693308#c2

### CODE REVIEW

**Comment by jklukas** [comment]
Needs the same logic applied to Fenix query.

**Comment by jklukas** [comment]
Marking @scholtzan as you reviewed the last PR for these tables (https://github.com/mozilla/bigquery-etl/pull/1056)

File: sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql [hunk 1]
```diff
@@ -52,9 +64,13 @@ SELECT
...
     SELECT AS STRUCT
       addon.id,
-      udf.mode_last(array_agg(addon.version)) AS version,
+      -- Same methodology as for app_version above.
+      ARRAY_AGG(addon.version ORDER BY mozfun.norm.truncate_version(addon.version, "minor") DESC)[
+        SAFE_OFFSET(0)
+      ] AS version,
     FROM
       UNNEST(addons) AS addon
+      WITH OFFSET AS i
```
Comment (scholtzan) [inline sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql:73]: `i` seems to be unused

**Review by scholtzan — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql (hunk 1 above)
Comment (jklukas) [reply sql/moz-fx-data-shared-prod/amo_prod/desktop_addons_by_client_v1/query.sql:73]: Will remove. This is a holdover from a previous variant of the design.

**Pull request merged**